src/index.js

```diff
@@ -146,12 +146,18 @@ export let ContextTree = _.curry(
         onError(error) // Raise the onError event
       }
     }
-    let triggerImmediateUpdate = F.debounceAsync(1, runUpdate)
-    let triggerDelayedUpdate = F.debounceAsync(debounce, runUpdate)
+
+    let triggerImmediatePathUpdate = _.memoize(() =>
+      F.debounceAsync(0, runUpdate)
+    )
+    let triggerDelayedPathUpdate = _.memoize(() =>
+      F.debounceAsync(debounce, runUpdate)
+    )
+
     let triggerUpdate = path =>
-      TreeInstance.disableAutoUpdate
-        ? triggerImmediateUpdate(path)
-        : triggerDelayedUpdate(path)
+      (TreeInstance.disableAutoUpdate
+        ? triggerImmediatePathUpdate
+        : triggerDelayedPathUpdate)(String(path))(path)
```

Comment:
What is `String(path)` for? if you're trying to do work around mobx observables, the way to do this is using `snapshot` from config (which is `toJS` in a mobx context)

Reply:
replaced with `encode`